### PR TITLE
Update thermostat behavior and update fork branch to be in line with current WyzeAccessory updates

### PR DESCRIPTION
**Pull request by carTloyal123**
This PR should address four things:

1. Refactor of WyzeAccessory to remove thermostat relevant components that were left previously
2. Update WyzeThermostat to hold all needed functionality inside it's own class and not rely on WyzeAccessory - this is to fall in line with the new structure of the project
3. Push a fix to an issue where you cannot set the thermostat temp above or below the arbitrary thresholds imposed by Homekit - #203 
4. Fix for constant reboot when using thermostat - #228

### CODE REVIEW

**Comment by carTloyal123** [comment]
@jfarmer08 feel free to take a look at this and give it a test locally. Works locally on my side but would appreciate a few other people trying it to make sure I did not break anything outside the thermostat. I only have two cameras and thermostat so not a great range of variables being tested.

File: src/accessories/WyzeThermostat.js [hunk 1]
```diff
@@ -78,31 +89,18 @@ module.exports = class WyzeThermostat extends WyzeAccessory {
...
   }
 
   async handleHeatingThresholdTemperatureGet() {
-    if (this.plugin.config.pluginLoggingEnabled)
-      this.plugin.log(
-        "[Thermostat] handleHeatingThresholdTemperatureGet Heat Setpoint: " +
-          this.thermostatHeatSetpoint
-      );
-
+    this.debugLog("handleHeatingThresholdTemperatureGet Heat Setpoint: " + this.thermostatHeatSetpoint);
```
Comment (jfarmer08) [inline src/accessories/WyzeThermostat.js:103]: I think this should be ```this.plugin.log```

File: src/accessories/WyzeThermostat.js [hunk 2]
```diff
@@ -78,31 +89,18 @@ module.exports = class WyzeThermostat extends WyzeAccessory {
...
   }
 
   async handleCoolingThresholdTemperatureGet() {
-    if (this.plugin.config.pluginLoggingEnabled)
-      this.plugin.log(
-        "[Thermostat] handleCoolingThresholdTemperatureGet Cool Setpoint: " +
-          this.thermostatCoolSetpoint
-      );
-
+    this.debugLog("handleCoolingThresholdTemperatureGet Cool Setpoint: " + this.thermostatCoolSetpoint);
```
Comment (jfarmer08) [inline src/accessories/WyzeThermostat.js:98]: I think this should be ```this.plugin.log``` Looks like a few other places also.

**Comment by jfarmer08** [comment]
The CHANGELOG.md log should reflect the change of this PR.

File: src/accessories/WyzeThermostat.js [hunk 3]
```diff
@@ -297,23 +228,122 @@ module.exports = class WyzeThermostat extends WyzeAccessory {
...
+  }
+
+  // Helper Functions
+
+  debugLog(message) {
+    if (this.plugin.config.logLevel == "debug" && this.plugin.config.pluginLoggingEnabled)
+      this.plugin.log.info(
+        this.colors.CYAN + "[Thermostat] " + message + this.colors.RESET
+      );
+  }
```
[the remark this answers is not in the document]
Comment (jfarmer08) [reply src/accessories/WyzeThermostat.js:345]: So is it changing the colors?

File: package.json [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
-  "version": "0.5.45",
+  "version": "0.5.46",
   "description": "Wyze Smart Home plugin for Homebridge",
   "license": "MIT",
   "main": "src/index.js",
   "scripts": {
     "lint": "eslint src/**.js --max-warnings=0",
-    "start": "homebridge -I -P . -U ~/Work/homebridge-wyze-api"
-
+    "start": "homebridge -I -P . -U . > homebrfidge.log"
```
Comment (jfarmer08) [inline package.json:9]: should this be homebridge.log?

**Comment by carTloyal123** [comment]
@jfarmer08 any other notes on this?

**Comment by jfarmer08** [comment]
I don't have a thermostat so I cannot test that, but I didn't test with my 83 other products and didn't see any issue.

**Comment by carTloyal123** [comment]
@jfarmer08 do you actually have 83 wyze products or you just have a bunch?

**Comment by jfarmer08** [comment]
Yeah I have about that many.

**Pull request merged**